Pcap++/header/PcapFilter.h

```diff
@@ -124,6 +119,37 @@ namespace pcpp
 		/// could not be compiled
 		bool matchPacketWithFilter(const uint8_t* packetData, uint32_t packetDataLength, timespec packetTimestamp,
 		                           uint16_t linkType);
+
+		/// @brief Match a packet with the filter stored in this object.
+		///
+		/// If the filter is empty the method returns "true".
+		/// If the link type of the raw packet is different than the one set in setFilter() the filter will be
+		/// recompiled.
+		///
+		/// @param[in] rawPacket The raw packet to match the filter against
+		/// @return True if the filter matches (or if it's empty). False otherwise
+		bool matches(const RawPacket& rawPacket) const;
+
+		/// @brief Match a raw buffer of packet data against the filter stored in this object.
+		///
+		/// If the filter is empty the method returns "true".
+		/// If the link type provided is different than the one set in setFilter() the filter will be recompiled.
+		///
+		/// @param[in] packetData A pointer to the raw packet data
+		/// @param[in] packetDataLength The length of the raw packet data in bytes
+		/// @param[in] timestamp Timestamp to be associated with the packet
+		/// @param[in] linkType The link type of the packet
+		/// @return True if the filter matches (or if it's empty). False otherwise
+		bool matches(const uint8_t* packetData, uint32_t packetDataLength, timespec timestamp, uint16_t linkType) const;
```

Comment:
ditto: if these are overloads of `matchPacketWithFilter` let's keep the same name?
We can also add `const` qualifier to the existing methods